src/Behat/Gherkin/Node/ExampleNode.php

```diff
@@ -87,13 +93,17 @@ public function getKeyword()
     }
 
     /**
-     * Returns example title.
+     * Returns the example row as a single string.
      *
      * @return string
+     *
+     * @deprecated You should normally not depend on the original row text, but if you really do, just switch to {@see self::getExampleText()}.
      */
     public function getTitle()
     {
-        return $this->title;
+        @trigger_error('`ExampleNode::getTitle()` calls should be migrated to `ExampleNode::getScenarioTitle()` or `ExampleNode::getExampleText()`.', E_USER_DEPRECATED);
```

Comment:
*if* we stick with deprecating this, I wouldn't trigger a runtime deprecation for this, just mark it for IDE's to highlight.

Reply:
That would make it harder for the end users to realise that (they would need to be actively looking at the usages). Are you still fine with that?

Reply:
Yeah, because most of the time I think it isn't / should not be a concern for an end user. 

If I am running behat with a third-party formatter, and my third-party formatter is calling a deprecated behat method, that shouldn't concern me as long as the formatter is still giving the output I expect. Assuming the formatter specifies its behat dependencies correctly, either I will eventually upgrade to a newer formatter release that solves it without me noticing, or I will be prevented from upgrading to a newer behat until the formatter is compatible.

That situation (and others like it) feel a lot more common than ones where the user themselves is calling the methods - and even if they are, it's not important that they do anything about it if they're happy with the titles their current code is displaying. Until / unless we remove it in a future BC break, but at that point they will find and fix it when they try to upgrade.

Reply:
It shouldn't affect the end user, however (unless they're capturing all deprecations indiscriminately). I thought this was handled automatically in the symfony deprecations helper for example; that deprecations would be shown in the source project and usages by its immediate dependants (but not further).

Anyway I don't have a strong opinion - I'm also fine without this.